tests/test_flows.py

```diff
@@ -436,36 +436,83 @@ def test_channel_selection(self):
             "{0x82 {0x00 0x00}}"
             )
 
-        debug("Send channel selection request")
-        mid = self.gateway_ucc.dev_send_1905(
-                self.mac_repeater1,
-                0x8006,
-                tlv(0x8B, 0x005F, "{} {}".format(self.mac_repeater1_wlan0, payload_wlan0)),
-                tlv(0x8D, 0x0007, "{} {}".format(self.mac_repeater1_wlan0, payload_empty)),
-                tlv(0x8B, 0x004C, "{} {}".format(self.mac_repeater1_wlan2, payload_wlan2)),
-                tlv(0x8D, 0x0007, "{} {}".format(self.mac_repeater1_wlan2, payload_empty))
+        """
+        Step 1: Trigger channel selection to channel 6 and 36. Check that
+                operating channel report was sent.
+
+        Step 2: Trigger channel selection to channel 6 and 36 again - check that
+                operating channel report is sent again. This is to catch bugs
+                when we don't send channel report when there is no need to
+                switch channel
+        """
+        for i in range(1, 3):
+            debug("Send channel selection request, step {}".format(i))
+            mid = self.gateway_ucc.dev_send_1905(
+                    self.mac_repeater1,
+                    0x8006,
+                    tlv(0x8B, 0x005F, '{} {}'.format(self.mac_repeater1_wlan0, payload_wlan0)),
+                    tlv(0x8D, 0x0007, '{} 0x{:2x}'.format(self.mac_repeater1_wlan0, tp20dBm)),
```

Comment:
This must be 
```suggestion
                    tlv(0x8D, 0x0007, '{} 0x{:02x}'.format(self.mac_repeater1_wlan0, tp20dBm)),
```

It happens to work now because we use values > 16 so it already is two digits, but if it weren't, the leading 0 is really essential.